src/microbe_stage/Microbe.cs

```diff
@@ -567,6 +567,31 @@ public override void _IntegrateForces(PhysicsDirectBodyState physicsState)
         }
     }
 
+    /// <summary>
+    ///   Returns a list of tuples, representing all possible compound targets. These are not all clouds that the
+    ///   microbe can smell; only the best candidate of each compound type.
+    /// </summary>
+    /// <param name="clouds">CompoundCloudSystem to scan</param>
+    /// <returns>
+    ///   A list of tuples. Each tuple contains the type of compound, the color of the line (if any needs to be drawn),
+    ///   and the location where the compound is located.
+    /// </returns>
+    public List<(Compound Compound, Color Colour, Vector3 Target)> GetDetectedCompounds(CompoundCloudSystem clouds)
+    {
+        var detections = new List<(Compound Compound, Color Colour, Vector3 Target)>();
+        foreach (var (compound, range, minAmount, colour) in activeCompoundDetections)
+        {
+            var detectedCompound = clouds.FindCompoundNearPoint(Translation, compound, range, minAmount);
+
+            if (detectedCompound != null)
+            {
+                detections.Add((compound, colour, (Vector3)detectedCompound));
```

Comment:
```suggestion
                detections.Add((compound, colour, detectedCompound.Value));
```
Sorry, if this suggestion is wrong, I'm making this suggestion assuming the type here is `Vector3?`